pydicom/filewriter.py

```diff
@@ -414,6 +414,17 @@ def write_data_element(fp, data_element, encoding=default_encoding):
     if (hasattr(data_element, "is_undefined_length")
             and data_element.is_undefined_length):
         is_undefined_length = True
+        # valid pixel data with undefined length shall contain encapsulated
+        # data, e.g. sequence items - raise ValueError otherwise (see #238)
+        if data_element.tag == 0x7fe00010:  # pixel data
```

Comment:
I just thought it too ugly/unreadable - the condition has to be split into 3 lines:
``` python
if data_element.tag == 0x7fe00010:  # pixel data
            val = data_element.value
            if (fp.is_little_endian and val.startswith(b'\xfe\xff\x00\xe0') or
                not fp.is_little_endian and 
                val.startswith(b'\xff\xfe\xe0\x00')):
                raise ValueError('Pixel Data with undefined length must '
                                 'start with an an item tag')
```